### DISCUSSION TITLE
Randomizing presentation of trials

### DISCUSSION THREAD

**Discussion by Stajduhar-A** [jspsych/jsPsych#783] [Q&A] [answered]
I want to randomize my trials and I saw that you can randomize objects using the jsPsych.randomization() function... however, I have the type parameter in each trial. I noticed that the objects you have as examples don't have this parameter... is there any way to get rid of it in the trial objects, randomize, and still have it work without type being specified? I'm curious how to go about that. Thanks!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @Stajduhar-A,

You can randomize objects that have a type parameter. This would work, for example:

```js
var trial_1 = {
	type: 'html-keyboard-response',
	stimulus: 'Trial 1'
}

var trial_2 = {
	type: 'html-keyboard-response',
	stimulus: 'Trial 1'
}

var trial_3 = {
	type: 'html-keyboard-response',
	stimulus: 'Trial 1'
}

var trial_order = jsPsych.randomization.shuffle([trial_1, trial_2, trial_3]);

jsPsych.init({
	timeline: trial_order
});
```

You may also want to look into [timeline variables](https://www.jspsych.org/overview/timeline/#timeline-variables), which provide a rich set of tools for randomizing trials.

**Reply by Stajduhar-A** [reply]
Thank you so much, Josh!

**Comment by SaoirseConnorDesai** [comment]
I have a further question about the solution above for randomizing objects that have a type parameter. If I wanted to present another trial or some instructions before the trial without randomizing these, how would I go about this?

For example, if I wanted to have 'instructions' first and then randomize 'trial_1', 'trial_2', 'trial_3' how would I specify this as the timeline variable in the jsPsych.init function?

I tried accessing the 'timeline variables' link but it says 'Page not found'.

**Reply by jodeleeuw** [reply]
I think this is now answered here https://github.com/jspsych/jsPsych/discussions/941#discussioncomment-47117. Adding this here to make finding the answer easier for future searches.
